src/ray/core_worker/task_manager.cc

```diff
@@ -71,7 +83,7 @@ void TaskManager::CompletePendingTask(const TaskID &task_id,
     pending_tasks_.erase(it);
   }
 
-  RemovePlasmaSubmittedTaskReferences(spec);
+  RemoveFinishedTaskReferences(spec, worker_addr, reply.borrowed_refs());
```

Comment:
Maybe `UpdateFinishedTaskReferences` or similar? Given that we aren't actually removing the ones that were borrowed.